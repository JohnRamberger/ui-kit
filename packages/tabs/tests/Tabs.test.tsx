import React from "react";

import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Tabs, TabItem, TabTitle } from "../";

describe("Tabs", () => {
  it("renders default", () => {
    const component = mount(
      <Tabs selectedIndex={0} onSelect={() => jest.fn()}>
        <TabItem>
          <TabTitle>Tab 1 Name</TabTitle>
          <div>First tab Content</div>
        </TabItem>
        <TabItem>
          <TabTitle>Tab 2 Name</TabTitle>
          Second Tab Content
        </TabItem>
      </Tabs>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("renders vertically", () => {
    const component = mount(
      <Tabs selectedIndex={0} onSelect={() => jest.fn()} direction="vert">
        <TabItem>
          <TabTitle>Tab 1 Name</TabTitle>
          <div>First tab Content</div>
        </TabItem>
        <TabItem>
          <TabTitle>Tab 2 Name</TabTitle>
          Second Tab Content
        </TabItem>
      </Tabs>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
