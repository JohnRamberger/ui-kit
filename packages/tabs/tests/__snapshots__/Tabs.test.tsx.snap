// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders default 1`] = `
<Memo(Tabs)
  onSelect={[Function]}
  selectedIndex={0}
>
  <Tabs
    className="react-tabs css-14ccmid"
    data-cy="tabs"
    defaultFocus={false}
    defaultIndex={null}
    disableUpDownKeys={false}
    environment={null}
    focusTabOnClick={true}
    forceRenderTabPanel={false}
    onSelect={[Function]}
    selectedIndex={0}
  >
    <UncontrolledTabs
      className="react-tabs css-14ccmid"
      data-cy="tabs"
      disableUpDownKeys={false}
      environment={null}
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={0}
    >
      <div
        className="react-tabs css-14ccmid"
        data-cy="tabs"
        data-rttabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id=":r0:0"
              key=".$.0"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="panel:r0:0"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                data-rttab={true}
                id="tab:r0:0"
                role="tab"
                tabIndex="0"
              >
                Tab 1 Name
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id=":r0:1"
              key=".$.1"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="panel:r0:1"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                data-rttab={true}
                id="tab:r0:1"
                role="tab"
                tabIndex={null}
              >
                Tab 2 Name
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id=":r0:0"
          key=".1:$.0"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
        >
          <div
            aria-labelledby="tab:r0:0"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="panel:r0:0"
            role="tabpanel"
          >
            <div
              key=".0"
            >
              First tab Content
            </div>
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id=":r0:1"
          key=".1:$.1"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
        >
          <div
            aria-labelledby="tab:r0:1"
            className="react-tabs__tab-panel"
            id="panel:r0:1"
            role="tabpanel"
          />
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</Memo(Tabs)>
`;

exports[`Tabs renders vertically 1`] = `
<Memo(Tabs)
  direction="vert"
  onSelect={[Function]}
  selectedIndex={0}
>
  <Tabs
    className="react-tabs css-628men"
    data-cy="tabs"
    defaultFocus={false}
    defaultIndex={null}
    disableUpDownKeys={false}
    environment={null}
    focusTabOnClick={true}
    forceRenderTabPanel={false}
    onSelect={[Function]}
    selectedIndex={0}
  >
    <UncontrolledTabs
      className="react-tabs css-628men"
      data-cy="tabs"
      disableUpDownKeys={false}
      environment={null}
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={0}
    >
      <div
        className="react-tabs css-628men"
        data-cy="tabs"
        data-rttabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id=":r1:0"
              key=".$.0"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="panel:r1:0"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                data-rttab={true}
                id="tab:r1:0"
                role="tab"
                tabIndex="0"
              >
                Tab 1 Name
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id=":r1:1"
              key=".$.1"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="panel:r1:1"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                data-rttab={true}
                id="tab:r1:1"
                role="tab"
                tabIndex={null}
              >
                Tab 2 Name
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id=":r1:0"
          key=".1:$.0"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
        >
          <div
            aria-labelledby="tab:r1:0"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="panel:r1:0"
            role="tabpanel"
          >
            <div
              key=".0"
            >
              First tab Content
            </div>
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id=":r1:1"
          key=".1:$.1"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
        >
          <div
            aria-labelledby="tab:r1:1"
            className="react-tabs__tab-panel"
            id="panel:r1:1"
            role="tabpanel"
          />
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</Memo(Tabs)>
`;
